### PR TITLE
Add exception to request trace error logging

### PR DESCRIPTION
**Pull request by johnlcox**
What was changed? Why is this necessary?
----------------------------------------

Added the `JsonProcessingException` to the error log message if serializing the request trace fails to serialize to JSON.

How was it tested?
----

Added a unit test to confirm the exception is logged out.

How to test
----

*This is bare minimum acceptable testing*

- [ ] `mvn clean install -U`

Reviewers
----

- [ ] [John Leacox](https://github.com/johnlcox)
- [ ] [Sundeep Paruvu](https://github.com/sparuvu)
- [ ] [Nimesh Subramanian](https://github.com/nimeshsubramanian)
- [ ] [Supriya Lal](https://github.com/lal-s)
- [ ] [Brian van de Boogaard](https://github.com/b-boogaard)

### CODE REVIEW

File: stagemonitor/src/main/java/com/cerner/beadledom/stagemonitor/request/LogJsonRequestTraceReporter.java [hunk 1]
```diff
@@ -35,16 +31,20 @@
...
     this.logger = LoggerFactory.getLogger(LogJsonRequestTraceReporter.class);
+    this.errorLogger =
+        LoggerFactory.getLogger(LogJsonRequestTraceReporter.class.getSimpleName() + ".Error");
     this.objectMapper = objectMapper;
     this.requestMonitorPlugin = Stagemonitor.getPlugin(RequestMonitorPlugin.class);
     this.jsonRequestTraceLoggerPlugin = Stagemonitor.getPlugin(JsonRequestTraceLoggerPlugin.class);
   }
 
   //@VisibleForTesting
   LogJsonRequestTraceReporter(
```
Comment (b-boogaard) [inline stagemonitor/src/main/java/com/cerner/beadledom/stagemonitor/request/LogJsonRequestTraceReporter.java]: Am I right in saying, that this second constructor is only used during testing?

We don't need to worry about injecting this second logger in from anywhere do we?

Comment (johnlcox) [inline stagemonitor/src/main/java/com/cerner/beadledom/stagemonitor/request/LogJsonRequestTraceReporter.java]: Correct.

This class only gets created here: https://github.com/cerner/beadledom/blob/master/stagemonitor/src/main/java/com/cerner/beadledom/stagemonitor/LogJsonRequestTraceReporterProvider.java

And used here (to register with stagemonitor): https://github.com/cerner/beadledom/blob/master/stagemonitor/src/main/java/com/cerner/beadledom/stagemonitor/RequestMonitorLifecycle.java

File: stagemonitor/src/main/java/com/cerner/beadledom/stagemonitor/request/LogJsonRequestTraceReporter.java [hunk 2]
```diff
@@ -60,10 +60,14 @@ public void reportRequestTrace(ReportArguments reportArguments) throws Exception
...
-        // It's not valid JSON, this log should only contain JSON, so log to error logger instead
+      } catch (JsonProcessingException e) {
+        // Error writing call stack to JSON, this log should only contain JSON,
+        // so log to error logger instead.
         errorLogger
-            .error("Unable to serialize stack request trace to json. Request: {}", requestTrace);
+            .error(
+                "Unable to serialize stack request trace to json. Error: {} Request: {}",
+                e,
+                requestTrace);
```
Comment (b-boogaard) [inline stagemonitor/src/main/java/com/cerner/beadledom/stagemonitor/request/LogJsonRequestTraceReporter.java]: Is there anyway get can get an idea of the size of the `requestTrace`? If so, should we either truncate them or not log them past a certain size? If so, should we add a configuration item so consumers can toggle the max size? Or perhaps a configuration option to allow consumers to opt out of logging the `requestTrace` in error scenarios all together?

Comment (johnlcox) [inline stagemonitor/src/main/java/com/cerner/beadledom/stagemonitor/request/LogJsonRequestTraceReporter.java]: I think we could do that, but I'd rather take this as a first step and see if we need configuration options.

Comment (b-boogaard) [inline stagemonitor/src/main/java/com/cerner/beadledom/stagemonitor/request/LogJsonRequestTraceReporter.java]: Sounds good.

**Pull request merged**